src/main/kotlin/no/fdk/dataset_catalog/security/SecurityConfig.kt

```diff
@@ -38,7 +40,14 @@ open class SecurityConfig : WebSecurityConfigurerAdapter() {
     @Bean
     open fun jwtDecoder(properties: OAuth2ResourceServerProperties): JwtDecoder? {
         val jwtDecoder = NimbusJwtDecoder.withJwkSetUri(properties.jwt.jwkSetUri).build()
-        jwtDecoder.setJwtValidator(DelegatingOAuth2TokenValidator(AudienceValidator()))
+        jwtDecoder.setJwtValidator(
+            DelegatingOAuth2TokenValidator(
+                listOf(
+                    JwtTimestampValidator(),
+                    JwtIssuerValidator(securityProperties.fdkIssuer),
+                    JwtClaimValidator("aud") { aud: List<String> -> aud.contains("fdk-registration-api") }
```

Comment:
Har ikke AudienceValidator denne funksjonaliteten?

Reply:
Ja, men dette var mer smooth. Så AudienceValidator blir slettet og bruker denne istedet.

Reply:
Jo, men er smudere å bruke claimvalidator for å validere en enkelt claim

Reply:
Ikke enig med begge dere. Se på denne: https://github.com/jzheaux/spring-security-oauth2-resource-server/blob/master/oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/resourceserver/authentication/AudienceValidator.java

Hvis `AudienceValidator` får et sett med tilatte audiences, så er det samme funksjonalitet sånn som det er ment.

Reply:
Men hvorfor skal vi bytte ut en veldig lesbar one-liner som gjør det vi vil med en hel klasse? Skjønner ikke fordleen

Reply:
Usikker på hva slags repo du linker til, det ser ut som en fork av en eldre versjon?

Egendefinert AudienceValidator ble nødvendig når jeg oppdaterte spring-security tidligere i høst, kanskje de legger til Audience senere men atm har ikke ny spring-security en AudienceValidator.